### PR TITLE
OWLS-86060 - Enhance server life-cycle scripts to support stopping/starting admin server

### PR DESCRIPTION
**Pull request by ankedia**
Enhanced server life cycle scripts to support stopping and starting admin server. Modified integration tests to verify new behavior. 
Integration test run URL for ItServerStartPolicy - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4352/
Complete run - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4363/

### CODE REVIEW

File: kubernetes/samples/scripts/domain-lifecycle/helper.sh [hunk 1]
```diff
@@ -71,6 +71,25 @@ function getEffectivePolicy {
...
+#
+# Function to get effective start policy of admin server
+# $1 - Domain resource in json format
+# $2 - Return value containing effective server start policy
+#      Legal retrun values are "NEVER" or "IF_NEEDED" or "ALWAYS".
+#
+function getEffectiveAdminPolicy {
+  local domainJson=$1
+  local __effectivePolicy=$2
+  local __adminStartPolicy=""
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/domain-lifecycle/helper.sh:83]: Can __domainStartPolicy be declared local? Or will that not work?

**Review by tbarnes-us — APPROVED** [review]
I posted one minor comment about declaring a var local that you might want to take a look at - otherwise this looks good to me.

File: kubernetes/samples/scripts/domain-lifecycle/helper.sh (hunk 1 above)
Comment (ankedia) [reply kubernetes/samples/scripts/domain-lifecycle/helper.sh:83]: Done.

**Review by anpanigr — COMMENTED** [review]
oracle.weblogic.kubernetes.ItServerStartPolicy.testAdminServerRestart tests fails in parallel run ( https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4346/ )

**Comment by ankedia** [comment]
> oracle.weblogic.kubernetes.ItServerStartPolicy.testAdminServerRestart tests fails in parallel run ( https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4346/ )

That's an old run before my changes to integration test. Please see latest result at - 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4363/

**Review by anpanigr — COMMENTED** [review]
Can we mention that ( -k ) option is only applicable to manged server not for Administrative server OR give a message that ( -k) option is being ignored in (stop/start)Server.sh when the server chooses to be an  Administrative  server

**Review by rjeberhard — APPROVED** [review]

File: kubernetes/samples/scripts/domain-lifecycle/README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ The operator provides sample scripts to start up or shut down a specific Managed
...
+The `startServer.sh` script starts a WebLogic Server in a domain. For clustered Managed servers, it either increases the `spec.clusters[<cluster-name>].replicas` value for the Managed Server's cluster by `1` or updates the `spec.managedServers[<server-name>].serverStartPolicy` attribute of the domain resource or both as necessary. For the administration server, it updates the value of `spec.adminServer.serverStartPolicy` attribute of the domain resource. For non-clustered Managed servers, it updates the `spec.managedServers[<server-name>].serverStartPolicy` attribute of the domain resource. The script provides an option to keep the `spec.clusters[<cluster-name>].replicas` value constant for clustered servers. See the script `usage` information by using the `-h` option.
```
Comment (rosemarymarano) [inline kubernetes/samples/scripts/domain-lifecycle/README.md:10]: Managed servers -> Managed Servers
it either increases -> either it increases
administration server -> Administration Server
value of -> value of the
Managed servers -> Managed Servers

File: kubernetes/samples/scripts/domain-lifecycle/README.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ domain.weblogic.oracle/domain1 patched
...
+The `stopServer.sh` script shuts down a running WebLogic Server in a domain. For clustered Managed servers, it either decreases the `spec.clusters[<cluster-name>].replicas` value for the Managed Server's cluster by `1` or updates the `spec.managedServers[<server-name>].serverStartPolicy` attribute of the domain resource or both as necessary. For the administration server, it updates the value of `spec.adminServer.serverStartPolicy` attribute of the domain resource. For non-clustered Managed servers, it updates the `spec.managedServers[<server-name>].serverStartPolicy` attribute of the domain resource. The script provides an option to keep the `spec.clusters[<cluster-name>].replicas` value constant for clustered servers. See the script `usage` information by using the `-h` option.
```
Comment (rosemarymarano) [inline kubernetes/samples/scripts/domain-lifecycle/README.md:28]: Managed servers -> Managed Servers
it either increases -> either it increases
administration server -> Administration Server
value of -> value of the
Managed servers -> Managed Servers

**Review by rosemarymarano — COMMENTED** [review]
Few edits.

**Review by rosemarymarano — APPROVED** [review]

**Review by anpanigr — APPROVED** [review]
LGTM

**Pull request merged**